### ISSUE TITLE
DuplicateConditionIfChainCheck — new semantic check for duplicate conditions in if-else-if chains

### ISSUE THREAD

**Issue by AseemPrasad** [checkstyle/checkstyle#21670]
### Description
Checkstyle currently does not provide a check that detects duplicate conditional expressions across `if ... else if` chains.
### Proposed Check
`DuplicateConditionIfChainCheck`
The check should report `else if` conditional expressions whose AST structures are identical to any preceding condition in the same `if-else-if` chain.
### Violation Example
```java
class Test {
    void process(int value) {
        if (value > 0) {
            doSomething();
        } else if (value > 0) { // violation: duplicate condition
            doSomethingElse();
        } else if (value == 0) {
            doZero();
        }
    }
}
Expected violation:

Duplicate condition expression identical to the condition at line 3.

Non-Violation Example
java
class Test {
    void process(int value) {
        if (value > 10) {
            doHigh();
        } else if (value > 5) {
            doMedium();
        } else if (value > 0) {
            doLow();
        }
    }
}
What should NOT be flagged
Standalone if statements without else if branches.
Distinct conditional expressions in else if branches (e.g. x > 10 vs x > 5).
if statements nested inside the body of an else block (unless structured as an else if chain).
Expected Configuration
xml
<module name="Checker">
    <module name="TreeWalker">
        <module name="DuplicateConditionIfChain"/>
    </module>
</module>
Implementation Notes
Structural comparison on AST nodes (TokenTypes.EXPR) should be used rather than raw source code text comparison so formatting, whitespace, line breaks, and comments do not affect evaluation.
Middle branches of an if-else-if chain should be evaluated starting from the root LITERAL_IF node (ast.getParent().getType() != TokenTypes.LITERAL_ELSE).

**Comment by romani** [comment]
How can you prove that this is real problem, and not artificial?
If we go this way easily next issue will be on duplicate statements.

**Comment by AseemPrasad** [comment]
Thanks for the feedback..

The main difference here between duplicate `if` conditions and generic duplicate statements is that **a duplicate condition in an `else if` chain guarantees unreachable dead code**. If `if (cond)` is false, evaluating `else if (cond)` will *always* evaluate to false, rendering the entire `else if` block unreachable logic.

This bug can usually occur during code refactoring when a developer copies an existing `else if` block to add a new case, but forgets to update the condition (e.g. copying `else if (type == TYPE_A)` and failing to change `TYPE_A` to `TYPE_B`).

Because of this, standard static analysis rules in the Java ecosystem explicitly target this exact pattern:
- **SonarQube / SonarLint**: Rule [`S1862`](https://rules.sonarsource.com/java/RSPEC-1862/) ("Related 'if/else if' statements should not have the same condition") - categorized as a Code Smell / Bug.
- **IntelliJ IDEA**: Built-in inspection *"Duplicate condition in 'if' statement chain"*.

Unlike duplicate sequential statements (which are often valid separate actions), a duplicate `else if` condition is almost always an unintentional logic bug or dead code. Adding this check allows Checkstyle users to catch these copy-paste defects at build time.

**Comment by AseemPrasad** [comment]
@romani i have raised the pr #21671 on this issue..
thank you..
